google/cloud/google_cloud_cpp_common.cmake

```diff
@@ -147,6 +147,14 @@ target_link_libraries(
            absl::variant
            Threads::Threads
            OpenSSL::Crypto)
+# Take a dependency on the OpenTelemetry API, if it is available.
+find_package(opentelemetry-cpp CONFIG)
```

Comment:
I think we can fix this later, but this may create problems with vcpkg (and other package managers).  Basically, if somebody does:

```
vcpkg install opentelemetry-cpp
vcpkg install google-cloud-cpp
```

Then `google-cloud-cpp` has a dependency on `opentelemetry-cpp` that the package manager knows nothing about.  While if they do:

```
vcpkg remove opentelemetry-cpp
vcpkg install google-cloud-cpp
```

Then they get a different set of deps.  I think you want something like:

```
if (experimental-opentelemetry IN_LIST GOOGLE_CLOUD_CPP_ENABLE)
  find_package(opentelemetry-cpp CONFIG QUIET)
  if (opentelemetry-cpp_FOUND)
... ...
endif ()

Reply:
Hmmm, we have to fix this before the next release, or we will not be able to push this to vcpkg, and Conda will pick up the change automatically.

Reply:
Argh. Good point. It makes sense for customers to explicitly request the feature anyway.

This PR is currently harmless (we are not taking a dep on opentelemetry, yet, despite the comment saying otherwise). But I will slap the `do not merge` label on it anyway, and add the cmake feature in a separate PR.